ENT-2468: Use format strings with named arguments for translator context

tlhmerry9 opened the pull request.

ptoscano commented on src/rct/cert_commands.py:
```diff
@@ -43,8 +43,8 @@ def _create_cert(self):
             raise InvalidCLIOptionError(
-                    _("Unable to read certificate file '%s': %s") % (cert_file,
-                        ce))
+                    _("Unable to read certificate file '{cert}}': {ce}}".format(cert=cert_file,
```
extra `}` twice

tlhmerry9 replied:
removed extra `}`

ptoscano commented on src/rhsmlib/facts/dmiinfo.py:
```diff
@@ -97,7 +97,7 @@ def _read_dmi(self, func):
             return func()
         except Exception as e:
-            log.warn(_("Error reading system DMI information with %s: %s"), func, e)
+            log.warn(_("Error reading system DMI information with {func}}: {e}}".format(func=func, e=e)))
```
extra `}` twice

tlhmerry9 replied:
removed extra `}`

ptoscano reviewed (changes requested):
There is a common mistake done in all the conversions: you have to apply `format()` on the result of `_()`, not on the original string, otherwise they do not get translated.

tlhmerry9 commented on src/rct/manifest_commands.py:
```diff
@@ -267,8 +267,8 @@ def _print_products(self, zip_archive):
                 raise certificate.CertificateException(
-                        _("Unable to read certificate file '%s': %s") % (cert_file,
-                        ce))
+                        _("Unable to read certificate file '{cert}': {ce}}".format(cert=cert_file,
```
removed extra `}`

ptoscano commented:
This also needs to be rebased on top of current master, without merge commits.

tlhmerry9 commented:
> There is a common mistake done in all the conversions: you have to apply `format()` on the result of `_()`, not on the original string, otherwise they do not get translated.

I updated the way I added `.format()` to be `_().format()` and I also went through these files and replace the majority of the `"%s" % value`.  I left a couple and will be looking at them in a few minutes.

tlhmerry9 commented on src/subscription_manager/i18n_optparse.py:
```diff
@@ -143,5 +143,5 @@ def error(self, msg):
         self.print_usage(sys.stderr)
         #translators: arg 1 is the program name, arg 2 is the error message
-        print(_("%s: error: %s") % (self.get_prog_name(), msg))
+        print(_("{name}}: error: {msg}}".format(name=self.get_prog_name(), msg=msg)))
```
fixed but this file was deleted and will not be included in this PR

tlhmerry9 commented:
> This also needs to be rebased on top of current master, without merge commits.

This has been done and squashed to one commit

tlhmerry9 commented:
> This also needs to be rebased on top of current master, without merge commits.

I need to fix some tests before you re-review

ptoscano commented on src/rhsm_debug/debug_commands.py:
```diff
@@ -181,13 +182,13 @@ def _do_command(self):
                 sfm = SaferFileMove()
                 sfm.move(tar_file_path, final_path)
-                print(_("Wrote: %s") % final_path)
+                print(_("Wrote: {final_path}\\").format(final_path=final_path))
```
extra `\\`

tlhmerry9 replied:
This was a typo!  Fixed

ptoscano commented on src/subscription_manager/cache.py:
```diff
@@ -231,7 +231,7 @@ def load_status(self, uep, uuid, on_date=None):
             log.error("Bad identity, unable to connect to server")
             return None
-        # all of the abover are subclasses of ConnectionException that
+        # all of the above are subclasses of ConnectionException that
```
while it's a good typo fix, i'd do it in a separate PR (usually better avoid unrelated changes)

tlhmerry9 replied:
put back and made note to open a PR, and going to include the typo in migrate.py also

ptoscano commented on src/subscription_manager/cache.py:
```diff
@@ -424,7 +424,7 @@ def profile_reporting_enabled(self):
         # variable is completely unset.
         if 'SUBMAN_DISABLE_PROFILE_REPORTING' in os.environ and \
-            os.environ['SUBMAN_DISABLE_PROFILE_REPORTING'].lower() in ['true', '1', 'yes', 'on']:
+                os.environ['SUBMAN_DISABLE_PROFILE_REPORTING'].lower() in ['true', '1', 'yes', 'on']:
```
unrelated change

tlhmerry9 replied:
changed back

ptoscano commented on src/subscription_manager/cache.py:
```diff
@@ -706,10 +706,8 @@ def check_for_update(self):
-                    file=self.CACHE_FILE,
-                    err=err
-                ))
+                log.debug("Cache file {c_file} is corrupted: {err}".format(c_file=self.CACHE_FILE,
+                                                                           err=err))
```
unrelated (and unneeded IMHO) change

tlhmerry9 replied:
put back

ptoscano commented on src/subscription_manager/gui/about.py:
```diff
@@ -73,10 +73,10 @@ def __init__(self, parent, backend):
-                                           server_versions['candlepin'])
+        sub_man_version_label.set_markup(_("<b>{name} version:</b> {version}").format(
+            name="subscription manager", version=client_versions['subscription-manager']))
+        backend_version_label.set_markup(_("<b>subscription management service {version}:</b> {version}").format(
```
the first `version` is just a word, no need to replace it

tlhmerry9 replied:
I think I was thinking I forgot the {} around it.  changed it back

ptoscano commented on src/subscription_manager/gui/about.py:
```diff
@@ -119,7 +119,6 @@ def _rhsmcertd_on(self):
         fnull = open(os.devnull, "w")
         try:
             # if status == 0 then true
-
```
extra change

tlhmerry9 replied:
put back the extra line

ptoscano commented on src/subscription_manager/migrate/migrate.py:
```diff
@@ -385,12 +387,13 @@ def check_has_access(self, rpc_session, session_key):
                 # called when not needed.  If we see this error, someone has made a programming
                 # mistake.
-                raise Exception("No session key available.  Check that XMLRPC connection is being made with credentials.")
+                raise Exception("No session key available.  "
+                                "Check that XMLRPC connection is being made with credentials.")
```
unrelated/unneeded change

tlhmerry9 replied:
put back

ptoscano commented on src/subscription_manager/migrate/migrate.py:
```diff
@@ -945,13 +954,15 @@ def validate_options(options):
     if options.service_level and not options.auto:
         # TODO Need to explain why this restriction exists.
-        system_exit(os.EX_USAGE, _("The --servicelevel and --no-auto options cannot be used together."))
+        system_exit(os.EX_USAGE, _("The --service-level and --no-auto options cannot be used together."))
```
good change, however unrelated; better do it in an own PR

tlhmerry9 replied:
Changed back and made a note so I can open a PR

ptoscano reviewed (changes requested):
This is a good improvements, thanks. However, I have few notes:
- I'd avoid updating `keys.pot` in the same commit, as it makes the patches bigger; also, the updated pot files has few strings less, so I'd investigate that as a separate task
- adding named placeholders in translatable strings is a good thing; however, I'd be nice to actually have meaningful names for the placeholders when possible: instead of `e`, `ec`, `msg`, `fn`, `c_file`, etc, I'd use less shortened forms like `error`, `exception`, `message`, `function`, `command`, `certificate`, etc
- other than translatable strings, also helper strings were changed; I personally consider this a different task/story than changing UI strings, as IMHO code like `"%s.json" % name` is still better than the `.format()` syntax (and there is no need to reorder placeholders in non-UI strings)

tlhmerry9 commented:
> This is a good improvements, thanks. However, I have few notes:
> 
> * I'd avoid updating `keys.pot` in the same commit, as it makes the patches bigger; also, the updated pot files has few strings less, so I'd investigate that as a separate task
> * adding named placeholders in translatable strings is a good thing; however, I'd be nice to actually have meaningful names for the placeholders when possible: instead of `e`, `ec`, `msg`, `fn`, `c_file`, etc, I'd use less shortened forms like `error`, `exception`, `message`, `function`, `command`, `certificate`, etc
> * other than translatable strings, also helper strings were changed; I personally consider this a different task/story than changing UI strings, as IMHO code like `"%s.json" % name` is still better than the `.format()` syntax (and there is no need to reorder placeholders in non-UI strings)

> using a restored `keys.pot` file
> made the names used in the format strings to be more descriptive
> put the other non-translatable strings back the way they were (using %s)

ptoscano commented on src/subscription_manager/gui/registergui.py:
```diff
@@ -1523,10 +1523,10 @@ def __init__(self, reg_info, async_backend, parent_window):
         self._initialize_consumer_name()
         self.registration_tip_label.set_label("<small>%s</small>" %
-                                          get_branding().GUI_FORGOT_LOGIN_TIP)
+                                              get_branding().GUI_FORGOT_LOGIN_TIP)
```
extra indentation change

tlhmerry9 replied:
Done

ptoscano commented on src/subscription_manager/gui/registergui.py:
```diff
@@ -1523,10 +1523,10 @@ def __init__(self, reg_info, async_backend, parent_window):
 
         self.registration_header_label.set_label("<b>%s</b>" %
-                                             get_branding().GUI_REGISTRATION_HEADER)
+                                                 get_branding().GUI_REGISTRATION_HEADER)
```
extra indentation change

tlhmerry9 replied:
done

ptoscano commented on src/subscription_manager/gui/registergui.py:
```diff
@@ -1748,8 +1748,8 @@ def _on_default_button_clicked(self, widget):
         self.server_entry.set_text("%s:%s%s" % (base_config.DEFAULT_HOSTNAME,
-            base_config.DEFAULT_PORT,
-            base_config.DEFAULT_PREFIX))
+                                                base_config.DEFAULT_PORT,
+                                                base_config.DEFAULT_PREFIX))
```
extra indentation changes

tlhmerry9 replied:
Done

ptoscano commented on src/subscription_manager/gui/registergui.py:
```diff
@@ -1805,7 +1805,7 @@ def set_server_entry(self, hostname, port, prefix):
         else:
             self.server_entry.set_text("%s:%s%s" % (hostname,
-                                       port, prefix))
+                                                    port, prefix))
```
extra indentation change

tlhmerry9 replied:
Done

ptoscano commented on src/subscription_manager/gui/registergui.py:
```diff
@@ -2049,7 +2048,7 @@ def _subscribe(self, uuid, current_sla, dry_run_result, callback):
         try:
             log.debug("Binding to subscriptions at service level: %s" %
-                    dry_run_result.service_level)
+                      dry_run_result.service_level)
```
extra indentation change

tlhmerry9 replied:
Done

ptoscano commented on src/subscription_manager/migrate/migrate.py:
```diff
@@ -945,7 +950,9 @@ def validate_options(options):
             system_exit(os.EX_USAGE, _("The --activation-key and --environment options cannot be used together."))
         if options.destination_user or options.destination_password:
-            system_exit(os.EX_USAGE, _("The --activation-key option precludes the use of --destination-user and --destination-password"))
+            system_exit(os.EX_USAGE,
+                        _("The --activation-key option precludes the use of --destination-user and "
+                          " --destination-password"))
```
extra indentation change

tlhmerry9 replied:
Done

ptoscano reviewed (changes requested):
Looking good now! There is just one placeholder to fix, and few extra indentation changes to rollback.

tlhmerry9 commented on src/rct/cert_commands.py:
```diff
@@ -86,19 +86,19 @@ def __init__(self):
-        print(_("DER size: %db") % self._get_der_size(pem))
+        print(_("Type: {certificate_type}").format(certificate_type=type_to_string(cert)))
+        print(_("Version: {certificate_version}").format(certificate_version=cert.version))
+        print(_("DER size: {open_file}b").format(open_file=self._get_der_size(pem)))
```
reverted back

tlhmerry9 commented:
> Looking good now! There is just one placeholder to fix, and few extra indentation changes to rollback.

I made all of the changes, please review again :)

ptoscano commented on src/rct/cert_commands.py:
```diff
@@ -86,19 +86,19 @@ def __init__(self):
-        print(_("Version: %s") % cert.version)
+        print(_("Type: {certificate_type}").format(certificate_type=type_to_string(cert)))
+        print(_("Version: {certificate_version}").format(certificate_version=cert.version))
         print(_("DER size: %db") % self._get_der_size(pem))
```
this was fine to change btw, it needs a `{size}` or similar placeholder

tlhmerry9 replied:
I changed it back to use format with a `{size}` placeholder

ptoscano commented on src/subscription_manager/migrate/migrate.py:
```diff
@@ -945,7 +950,7 @@ def validate_options(options):
             system_exit(os.EX_USAGE, _("The --activation-key and --environment options cannot be used together."))
         if options.destination_user or options.destination_password:
-            system_exit(os.EX_USAGE, _("The --activation-key option precludes the use of --destination-user and --destination-password"))
+            system_exit(os.EX_USAGE, _("The --activation-key option precludes the use of --destination-user and  --destination-password"))
```
extra whitespace addition

ptoscano reviewed (changes requested):
:+1: very nice improvements, almost there

ptoscano commented:
:+1: LGTM now -- thanks for the work and the patience!
